packages/url_launcher/url_launcher_ios/ios/Classes/URLLauncherPlugin.swift

```diff
@@ -0,0 +1,96 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+
+public final class URLLauncherPlugin: NSObject, FlutterPlugin, UrlLauncherApi {
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let plugin = URLLauncherPlugin()
+    UrlLauncherApiSetup.setUp(binaryMessenger: registrar.messenger(), api: plugin)
+    registrar.publish(plugin)
+  }
+
+  private var currentSession: URLLaunchSession?
+  private let launcher: Launcher
+
+  private var topViewController: UIViewController? {
+    // TODO(stuartmorgan) Provide a non-deprecated codepath. See
+    // https://github.com/flutter/flutter/issues/104117
+    UIApplication.shared.keyWindow?.rootViewController?.topViewController
+  }
+
+  init(launcher: Launcher = UIApplicationLauncher()) {
+    self.launcher = launcher
+  }
+
+  func canLaunchUrl(url: String) -> LaunchResult {
+    guard let url = URL(string: url) else {
+      return .failedToLoad
+    }
+    let canOpen = launcher.canOpenURL(url)
+    return canOpen ? .success : .failedToLoad
+  }
+
+  func launchUrl(
+    url: String, universalLinksOnly: Bool,
+    completion: @escaping (Result<LaunchResult, Error>) -> Void
+  ) {
+    guard let url = URL(string: url) else {
+      completion(.success(.failedToLoad))
+      return
+    }
+    let options = [UIApplication.OpenExternalURLOptionsKey.universalLinksOnly: universalLinksOnly]
+    launcher.openURL(url, options: options) { result in
+      completion(.success(result ? .success : .failedToLoad))
+    }
+  }
+
+  func openUrlInSafariViewController(
+    url: String, completion: @escaping (Result<LaunchResult, Error>) -> Void
```

Comment:
line break. please go over all signatures and make sure comma separated items are in separate lines.